Zendesk documentation updates (#813)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/zendesk.mdx

<div class="infoBlock">
RudderStack assumes that an <code class="inline-code">identify</code> call (containing the <code class="inline-code">userId</code>) is made before any <code class="inline-code">group</code> call. 
</div>

If you do not want to associate a particular user with a group, enable the **Send Group Calls without User ID** dashboard setting. If this setting is enabled, the group will be created or updated only if `userId` is **not** present in the event.

### Standard mapping

RudderStack maps the following **optional** `group` traits to the standard Zendesk organization attributes: 

| **RudderStack** | **Zendesk**    | 
| :---------------| :------------- | 
| `name`          | `name`         |  
| `domainNames`   | `domain_names` |
| `tags`          | `tags`         |
| `groupId`       | `external_id`  |
| `url`           | `url`          |
| `deleted`       | `deleted`      |

<div class="infoBlock">
RudderStack creates a new custom field in Zendesk for any other unmapped attributes.
</div>

### How RudderStack does the user-group association

> If both `userId` and `groupId` are present in the `group` event, then RudderStack first finds the user using the `email` field present in the <Link to="/event-spec/standard-events/common-fields/#contextual-fields">`context.traits`</Link> object.

Reviewer: ```suggestion
If both `userId` and `groupId` are present in the `group` event, then RudderStack first finds the user by looking for the `email` field in the <Link to="/event-spec/standard-events/common-fields/#contextual-fields">`context.traits`</Link> object.
```

> - If both `userId` and `groupId` are present in the `group` event, then RudderStack first finds the user using the `email` field present in the <Link to="/event-spec/standard-events/common-fields/#contextual-fields">`context.traits`</Link> object.

Author: ```suggestion
- If both `userId` and `groupId` are present in the `group` event, then RudderStack first finds the user by looking for the `email` field present in the <Link to="/event-spec/standard-events/common-fields/#contextual-fields">`context.traits`</Link> object.
```

- If `email` is absent, RudderStack creates the user. It will then find the organization associated with the `groupId` present in the payload.

> - If `email` is present, then RudderStack associates the user with the `organizationId` and sends all the user information for this call.

Reviewer: reversing their order will look more connected, I suppose

Once you have verified that the source platform supports sending events to Zendesk, perform the steps below:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Zendesk**.
2. Assign a name to the destination and click **Continue**.

### Connection settings

To successfully configure Zendesk as a destination, you will need to configure the following settings:

<img src="../../assets/event-stream-destinations/zendesk-connection-settings.png" alt="Zendesk connection settings"/>

* **Email**: Enter the email address used to log into your Zendesk account.
* **API Token**: Enter the Zendesk API token used to authenticate the integration. To generate a new API token, refer to this <a href="https://support.zendesk.com/hc/en-us/articles/226022787-Generating-a-new-API-token-">Zendesk support article</a>.
* **Zendesk Subdomain**: Enter your Zendesk subdomain excluding `.zendesk.com`.

> * **Create users as verified**: Enable this setting to create verified users in Zendesk. If this setting is enabled, RudderStack sends `verified` as `true` in the event payload and Zendesk skips the email verification for these users.

Reviewer: ```suggestion
* **Create users as verified**: Enable this setting to create verified users in Zendesk. When enabled, RudderStack sends `verified` as `true` in the event payload and Zendesk skips the email verification for these users.
```

> * **Send Group calls without a User ID**: Enable this setting if you do not want to associate a particular user with a group. If this setting is enabled, the group will be created or updated only if `userId` is **not** present in the event.

Reviewer: ```suggestion
* **Send Group calls without a User ID**: Enable this setting if you do not want to associate a particular user with a group. When enabled, a group is created or updated only if a `userId` is **not** present in the event.
```

* **Enable Removing Users from Organizations**: Enable this setting to remove users from an organization via the `identify` call.

<div class="infoBlock">
For more information on this setting, refer to the <Link to="#removing-users-from-an-organization">Removing users from an organization</Link> section below.
</div>

## Identify

You can use the <Link to="/event-spec/standard-events/identify/">`identify`</Link> call to create or update a user in Zendesk. 

<div class="infoBlock">
It is highly recommended to include the user's <code class="inline-code">email</code> in the <code class="inline-code">identify</code> call as RudderStack uses this field to create or update a user in Zendesk.
</div>

```javascript
rudderanalytics.identify("1hKOmRA4GRlm", {
  name: "Alex Keener",
  email: "alex@example.com",
  country: "USA",
});
```

RudderStack persists the user details from the `identify` call for all the subsequent calls made to Zendesk.

### Supported mapping

> RudderStack maps the event properties to a subset of the standard Zendesk user attributes. The following table highlights the supported property mappings:

Reviewer: Dint quite get the 'subset' thing

Author: fixed the sentence - it was unwanted.

| **RudderStack property**  | **Zendesk attribute**       | **Presence** | 
| :--------------- | :---------------- | :------ |
| `name`           | `name`            | Required |
| `email`          | `email`           | Optional, but recommended | 
| `organizationId` | `organization_id` | Optional |
| `timezone`       | `time_zone`       |  Optional |
| `phone`          | `phone`           | Optional |
| `userId`         | `user_id`         | Optional |
| `userId`         | `external_id`     | Optional |

<div class="infoBlock">
RudderStack creates a new custom field in Zendesk for any other unmapped attributes.
</div>

<div class="infoBlock">
The <code class="inline-code">organization_id</code> corresponds to the organization ID assigned by Zendesk.
</div>

> ### Verifying users at creation

Reviewer: Can we do away with this section? As it provides similar info as in the connection setting

Author: Yep! Fixed this.

> The above `identify` call updates the user as well as removes them from the organization associated with the ID `900001329943`.

Reviewer: updates as well as removes is bit confusing.. should it be only remove?

Author: i think it means the updates the other user traits like `name`, `email`, `country`, etc. and also removes the user from the organization if `company.remove` is set to `true`.

RudderStack assumes a valid Zendesk `organization_id` in the `company.id` field. Then, it finds the organization corresponding to that ID and removes the user. 

## Track

> The <Link to="/event-spec/standard-events/track/">`track`</Link> call lets you record the user actions along with any properties associated with them and send this information to Zendesk.

Reviewer: ```suggestion
The <Link to="/event-spec/standard-events/track/">`track`</Link> call lets you record the user actions along with any associated properties and send this information to Zendesk.
```

You need to first set the `userId` by calling `identify` before sending any `track` events to Zendesk. 

<div class="infoBlock">

> Zendesk expects a <code class="inline-code">userId</code> for every <code class="inline-code">track</code> call.

Reviewer: we dont mention a user id in the sample call

Author: you're right - will work with @yashasvibajpai to fix this.
